projects/schematics/src/migrations/4_0/constructor-deprecations/data/cms-components.service.migration.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  ACTION_SUBJECT,
+  ANGULAR_CORE,
+  CMS_COMPONENTS_SERVICE,
+  CMS_FEATURES_SERVICE,
+  CONFIG_INITIALIZER_SERVICE,
+  FEATURE_MODULES_SERVICE,
+  OBJECT_TYPE,
+  PLATFORM,
+  PLATFORM_ID_STRING,
+  SPARTACUS_CORE,
+  SPARTACUS_STOREFRONTLIB,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const CMS_COMPONENTS_SERVICE_MIGRATION_1: ConstructorDeprecation = {
+  class: CMS_COMPONENTS_SERVICE,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [
+    {
+      className: PLATFORM,
+      literalInference: OBJECT_TYPE,
+      injectionToken: {
+        token: PLATFORM_ID_STRING,
+        importPath: ANGULAR_CORE,
+      },
+    },
+  ],
```

Comment:
In deprecatedParams CmsConfig is missing (in both cases). 
Also it would be good to have case when customers had constructor already with all 4 parameters.

Reply:
fixed